contrib/mesos/pkg/scheduler/queuer/queuer.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queuer
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	log "github.com/golang/glog"
+	"k8s.io/kubernetes/contrib/mesos/pkg/queue"
+	"k8s.io/kubernetes/contrib/mesos/pkg/runtime"
+	annotation "k8s.io/kubernetes/contrib/mesos/pkg/scheduler/meta"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+)
+
+const (
+	enqueuePopTimeout  = 200 * time.Millisecond
+	enqueueWaitTimeout = 1 * time.Second
+	yieldPopTimeout    = 200 * time.Millisecond
+	yieldWaitTimeout   = 1 * time.Second
+)
+
+type Queuer struct {
+	lock            sync.Mutex       // shared by condition variables of this struct
+	podUpdates      queue.FIFO       // queue of pod updates to be processed
+	PodQueue        *queue.DelayFIFO // queue of pods to be scheduled
```

Comment:
I'd rather see this exposed via some `Queue()` func instead of a read/write field in the struct